test/test_re/test_correlated_field.py

```diff
@@ -203,3 +203,107 @@ def test_nifty_vs_niftyre_matern_cf(
     npos = {k: ift.makeField(cf.domain[k], v) for k, v in pos.items()}
     npos = ift.MultiField.from_dict(npos, cf.domain)
     assert_allclose(cf(npos).val, jcf(pos))
+
```

Comment:
Note to self; before merging, format file and maybe summarize `add_fluctuations` calls.

To avoid duplicate code, maybe put `add_fluctuations` into a loop and simply join all parameters to `test_nifty_vs_niftyre_product` into e.g. a list of dicts `sdt = (dict(shape=(4,), distance=None, harmonic_type="spherical"), dict(shape=(3,), distance=0.1, harmonic_type="fourier"))`.

Reply:
I swapped the `...1` and `...2` code with a loop and merged `test_nifty_vs_niftyre_spherical` into the general CFM test.